test/e2e/testdata/template-failed-metric.yaml

```
@@ -0,0 +1,54 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: dag-task-test
+spec:
+  entrypoint: dag-task
+  templates:
+  - name: dag-task
+    metrics:
+        prometheus:
+        - counter:
+            value: "1"
+          # help: DAG Task Workflow succeeded
+          # name: dagtask_workflow_success
+          # when: '{{status}} == Succeeded'
+          help: DAG Task Workflow failed
+          name: dagtask_workflow_failure
+          when: '{{status}} == Failed'
+    dag:
+      tasks:
+      - name: TEST-ONE
+        template: echo
+        arguments:
+          parameters: 
+            - name: message
+              value: "console output-->TEST-{{item.command}}"
+            - name: tag
+              value: "{{item.tag}}"
+        withItems:
+          - { tag: TEST-ONE-A, command: ONE-A }
+          - { tag: TEST-ONE-B, command: ONE-B }
```

```suggestion
      - name: test
        template: echo
        arguments:
          parameters: 
            - name: message
              value: "{{item.message}}"
        withItems:
          - { message: testA }
          - { message: testB }
```

I don't think the "tag" is necessary, and the "command" looks like a misnomer to me.

In fact I'm not sure you really need a loop at all:
```suggestion
      - name: test
        template: echo
        arguments:
          parameters: 
            - name: message
              value: "test"
```

---

This workflow just comes from the repo itself, it's just a DAG with a metric. It would be better to simplify it like this, will do.